projects/lights_control/src/lights_blinker.c

```diff
@@ -0,0 +1,64 @@
+#include <stddef.h>
+
+#include "event_queue.h"
+#include "soft_timer.h"
+#include "status.h"
+
+#include "lights_blinker.h"
+
+static void prv_timer_callback(SoftTimerID timer_id, void *context) {
+  LightsBlinker *blinker = (LightsBlinker *)context;
+  blinker->state = !blinker->state;
+  event_raise(blinker->event_id, (uint16_t)blinker->state);
+  soft_timer_start_millis(blinker->duration, prv_timer_callback, (void *)blinker,
+                          &blinker->timer_id);
+}
+
+// blinker's timer id needs to be initialized to an invalid timer if it's not being used otherwise
+// we may have a collision.
+StatusCode lights_blinker_init(LightsBlinker *blinker) {
+  blinker->timer_id = SOFT_TIMER_INVALID_TIMER;
+  return STATUS_CODE_OK;
+}
+
+StatusCode lights_blinker_on(LightsBlinker *blinker, LightsBlinkerDuration duration, EventID id) {
+  if (lights_blinker_inuse(blinker)) {
+    // the passed-in blinker may have active timers
+    return STATUS_CODE_INVALID_ARGS;
+  }
+  blinker->duration = duration;
+  blinker->state = LIGHTS_BLINKER_STATE_ON;
+  blinker->event_id = id;
+  status_ok_or_return(event_raise(blinker->event_id, (uint16_t)blinker->state));
+  return soft_timer_start_millis(blinker->duration, prv_timer_callback, (void *)blinker,
+                                 &blinker->timer_id);
+}
+
+StatusCode lights_blinker_off(LightsBlinker *blinker) {
+  blinker->state = LIGHTS_BLINKER_STATE_OFF;
+  status_ok_or_return(event_raise(blinker->event_id, (uint16_t)blinker->state));
+  // we're expecting this to always return true, since every time the timer goes off, a new timer
+  // id gets set.
+  if (soft_timer_cancel(blinker->timer_id)) {
+    blinker->timer_id = SOFT_TIMER_INVALID_TIMER;
+    return STATUS_CODE_OK;
+  }
+  return STATUS_CODE_INTERNAL_ERROR;
```

Comment:
Please check all return codes.

Reply:
This was already fixed.

Reply:
Check again.

Reply:
Sorry, but I think you're looking at an older revision. This comment itself is marked as "outdated" on github:
<img width="842" alt="screen shot 2018-05-04 at 12 26 28 pm" src="https://user-images.githubusercontent.com/10492324/39639145-6fa73520-4f96-11e8-8ded-a7541b3818b6.png">

Reply:
Check all status codes in your code. I'm not going to point out every single place you need to fix this.

Reply:
I'll strictly be checking for that before every commit. Sorry for the inconvenience.